tests/core/test_core.py

```diff
@@ -279,23 +281,16 @@ def test_task_get_template(self):
         assert context['ds'] == '2015-01-01'
         assert context['ds_nodash'] == '20150101'
 
-        # next_ds is 2015-01-02 as the dag interval is daily
+        # next_ds is 2015-01-02 as the dag schedule is daily.
         assert context['next_ds'] == '2015-01-02'
         assert context['next_ds_nodash'] == '20150102'
 
-        # prev_ds is 2014-12-31 as the dag interval is daily
-        assert context['prev_ds'] == '2014-12-31'
-        assert context['prev_ds_nodash'] == '20141231'
-
         assert context['ts'] == '2015-01-01T00:00:00+00:00'
         assert context['ts_nodash'] == '20150101T000000'
         assert context['ts_nodash_with_tz'] == '20150101T000000+0000'
 
-        assert context['yesterday_ds'] == '2014-12-31'
-        assert context['yesterday_ds_nodash'] == '20141231'
-
-        assert context['tomorrow_ds'] == '2015-01-02'
-        assert context['tomorrow_ds_nodash'] == '20150102'
+        assert context['data_interval_start'].isoformat() == '2015-01-01T00:00:00+00:00'
+        assert context['data_interval_end'].isoformat() == '2015-01-02T00:00:00+00:00'
```

Comment:
Let's add test for deprecated configs too

Reply:
If the DAG is expected to be scheduled for each midnight (say `0 0 * * *`), data intervals always start and end at midnight for all scheduled (and backfilled) runs, so a manual run should match that. If the interval is `timedelta(days=1)` then yes the interval for this manual run should be between 1:00 24th and 25th. I will modify the docstring to clarify this.

Also see discussion in https://github.com/apache/airflow/pull/16352#discussion_r678354195 where Ash proposes a different data interval 😄